### PR TITLE
bonnie-prior 1302 nokogiri/loofah/rails-html-sanitizer fixes

### PR DESCRIPTION
**Pull request by c-monkey**
This PR fixes vulnerabilities found running `bundle exec bundle-audit check --update`. It also adds `bundle-audit` to travis to identify vulnerabilities that arise in future PRs.

----

:exclamation: **Must be updated before merge to point `patientapi` to bonnie-prior branch and `HDS` to mongoid5 branch. These lines are in the Gemfile, commented-out.**
- [x] [HDS](https://github.com/projectcypress/health-data-standards/pull/566)
- [x] [patientapi](https://github.com/projecttacoma/patientapi/pull/20)

----

Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [X] This pull request describes why these changes were made.
- [X] This PR is into the correct branch.
- [X] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1302
- [x] JIRA ticket links to this PR
- [X] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [X] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [X] Automated regression test(s) pass

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,202 +1,241 @@
...
+    bson (4.3.0)
+    builder (3.2.3)
+    bundler-audit (0.6.0)
+      bundler (~> 1.2)
+      thor (~> 0.18)
+    coderay (1.1.2)
+    coffee-rails (4.2.2)
       coffee-script (>= 2.2.0)
-      railties (>= 4.0.0, < 5.0)
+      railties (>= 4.0.0)
```
Comment (lukoktonos) [inline Gemfile.lock:77]: coffee-rails currently has the following dependency on railties, so I think this should be `railties (>= 4.0.0, <5.1)`:
image

File: hqmf2js.gemspec [hunk 2]
```diff
@@ -9,14 +9,9 @@ Gem::Specification.new do |s|
...
-#  s.add_dependency 'tilt', '~> 1.3.3'
-#  s.add_dependency 'coffee-script', '~> 2.2.0'
-#  s.add_dependency 'sprockets', '~> 2.2.2'
-#  s.add_development_dependency "awesome_print", "~> 1.1.0"
-#  s.add_dependency 'health-data-standards', '~> 3.1.1'
-#  s.add_dependency 'hquery-patient-api', '~> 1.0.2'
-
   s.files = s.files = `git ls-files`.split("\n")
+  s.add_dependency "rails", "~> 4.2.7"
+  s.add_dependency 'health-data-standards', '~>3.7'
```
Comment (lukoktonos) [inline hqmf2js.gemspec]: `'~>3.7'` -> `'~> 3.7'`

**Review by lukoktonos — APPROVED** [review]

File: Gemfile.lock [hunk 3]
```diff
@@ -1,202 +1,241 @@
...
     rack-test (0.6.3)
       rack (>= 1.0)
-    rails (4.1.12)
-      actionmailer (= 4.1.12)
-      actionpack (= 4.1.12)
-      actionview (= 4.1.12)
-      activemodel (= 4.1.12)
-      activerecord (= 4.1.12)
-      activesupport (= 4.1.12)
+    rails (4.2.10)
```
Comment (hossenlopp) [inline Gemfile.lock:136]: This is updating rails past where we have it in bonnie at 4.2.8. This does lead me to ask the question should we be updating rails and friends to 4.2.10 across the board?

File: Gemfile.lock (hunk 1 above)
Comment (c-monkey) [reply Gemfile.lock:77]: @losborne and I looked at this and reproduced it on his machine; for some reason bundler doesn't add this restriction to coffee-rails dependencies. This removal wasn't hand-jammed. Not going to mess with it.

File: Gemfile.lock (hunk 3 above)
Comment (c-monkey) [reply Gemfile.lock:136]: @hossenlopp and I looked at all the Gemfiles, they all state "~> 4.2.7" and this Gemfile.lock statement doesn't set a requirement, so the strange bundler behavior we think isn't problematic.

**Review by hossenlopp — APPROVED** [review]
